sdk/python3/ccnp/quote/quote_sdk.py

```diff
@@ -0,0 +1,350 @@
+    """Quote client class
+
+    This class is a client to connect to Quote Server and do gRPC call getting the
+    server.
+
+    Attributes:
+        _server (str): The gRPC server to connect.
+        _channel (Channel): The gRPC channel, thread-safe.
+        _nonce (str): The nonce parameter to get quote.
+        _user_data (str): The user data parameter to get quote.
+        _stub (GetQuoteStub): The get quote stub for gRPC.
+        _request (GetQuoteRequest): The get quote request for gRPC.
+    """
+    def __init__(self, server: str="unix:/run/ccnp/uds/quote-server.sock"):
+        """Initialize a quote client object
+
+        This constructor initializes quote client object with Unix Domain Socket (UDS)
+        path. And prepare default atrributes.
+
+        Args:
+            server (str): gRPC server UDS path, default is /run/ccnp/uds/quote-server.sock
+        
+        Raises:
+            ValueError: If server UDS path is not valid.
+        """
+        if len(server) == 0 or server[:5] != "unix:":
+            raise ValueError("Invalid server path, only unix domain socket supported.")
+        self._server = server
+        self._channel = None
+        self._stub = None
+        self._request = None
+
+    def request(self, nonce: str, user_data: str) -> Optional[quote_server_pb2.GetQuoteResponse]:
+        """Do reuqest to Quote Server
+        Detect the Quote Server and gRPC connect to the server. Make the getting quote stub
+        and request for communication.
+
+        Args:
+            nonce (str): The nonce parameters for getting quote.
+            user_data (str): The user data parameters for getting quote.
+
+        Raises:
+            RuntimeError: If Quote Server does not start.
+            ConnectionRefusedError: If connect to Quote Server failed.
+        """
+        if self._channel is None:
+            if not os.path.exists(self._server.replace('unix:', '')):
+                raise RuntimeError("Quote server does not start.")
+            self._channel = grpc.insecure_channel(self._server)
+            try:
+                grpc.channel_ready_future(self._channel).result(timeout=TIMEOUT)
+            except grpc.FutureTimeoutError as err:
+                raise ConnectionRefusedError('Connection to quote server failed') from err
+            self._stub = quote_server_pb2_grpc.GetQuoteStub(self._channel)
+
+        self._request = quote_server_pb2.GetQuoteRequest(nonce=nonce, user_data=user_data)
+        resp = self._stub.GetQuote(self._request)
+        return resp
+
+class Quote(ABC):
+    """An abstract base class for Quote
+
+    This class a abstract class with a common static method `get_quote` for external
+    SDK interface, the subclasses need to implement `parse` method to parse Quote
+    information.
+
+    Attributes:
+        _quote (bytes): The bytes of a quote.
+        _type (str): The type of a quote.
+    """
+
+    def __init__(self, quote: str, quote_type: str):
```

Comment:
One worry here. static method can be called thru class itself or instance of the class. But the definition here prohibits the usage of calling the get_quote() function with an instance of Quote() class, as there's no quote available while doing init.

Reply:
> One worry here. static method can be called thru class itself or instance of the class. But the definition here prohibits the usage of calling the get_quote() function with an instance of Quote() class, as there's no quote available while doing init.

This can be called by flow: Quote.get_quote()->QuoteTDX.__init__()->Quote.__init__()

Reply:
Yes. I understand the usage you mentioned. I mean (just in case!) that user call the api like this, it would fail:
quote_instance = Quote() --> fail with Value error
quote = quote_instance.get_quote()

Reply:
Yes, you are right, I will update the implementation for this.

Reply:
Done.